lib/register.coffee

```diff
@@ -78,21 +77,23 @@ exports.register = Promise.method (options, callback) ->
 	for opt in [ 'userId', 'applicationId', 'uuid', 'deviceType', 'provisioningApiKey', 'apiEndpoint']
 		if !options[opt]?
 			throw new Error("Options must contain a '#{opt}' entry.")
-	postAsync
+	fetch("#{options.apiEndpoint}/device/register?apikey=#{options.provisioningApiKey}"
 		method: 'POST'
-		url: "#{options.apiEndpoint}/device/register?apikey=#{options.provisioningApiKey}"
-		timeout: 30000
-		gzip: true
-		json: true
-		body:
+		headers:
+			'accept': 'application/json'
+			'accept-encoding': 'gzip'
+			'content-type': 'application/json'
+		body: JSON.stringify
 			user: options.userId
 			application: options.applicationId
 			uuid: options.uuid
 			device_type: options.deviceType
 			api_key: options.deviceApiKey
-	.tap ([{ statusCode }, body]) ->
-		if statusCode isnt 201
-			throw new Error(body)
-	.get(1)
+	).timeout(30000)
```

Comment:
I think for this specific module it's ok because it will probably never make many requests, but leaving sockets/requests hanging around in a server context is generally pretty bad as it is possible to get enough to exhaust resources and break things (we have had that happen) - so it'd be good to have it as a standard best practice thing across all our uses of fetch, and request for that matter